packages/controls/src/controls/combobox/combobox.ts

```diff
@@ -114,6 +120,18 @@ class Definition<C extends Config> extends ControlDefinition<
     return data
   }
 
+  resolveValue(
+    data: DataType<C> | undefined,
+  ): Resolvable<ResolvedValueType<C> | undefined> {
+    return {
+      readStableValue: (previous?: ResolvedValueType<C>) => {
+        const r = this.fromData(data)?.value
+        return deepEqual(r, previous) ? previous : r
```

Comment:
That being said, this might be no worse than the current implementation with `<RenderHook>`? I don't remember using `deepEqual` there but I guess an equivalent comparison would be to see if `<RenderHook>` re-renders whenever there's a prop change even if it wasn't that particular `<RenderHook>'s prop

Reply:
This is the only control that relies on deep comparison because the combobox data can literally be anything.

> A way to avoid this would be to store the last resolved value in state

That (store the last resolved value in state) is what we do in the upcoming PR that is stacked on top of this one, which is then passed here as `previous`. That said, I don't see how this eliminates the need for comparison if we want a stable value.

Reply:
Because we only would recompute the state if the input data changes, which we can check with triple equality, without a deep equals

Reply:
>  we only would recompute the state if the input data changes, which we can check with triple equality, without a deep equals

Okay, I'll have to see how this converges in the follow-up PR. I removed `deepEqual` here for now, making it equivalent to the old code this is replacing: https://github.com/makeswift/makeswift/pull/814/files#diff-ecff4567a2065c101d4623ba2045c46bb42572e4725fec3583b850a894db4bb5